services/twilio.js

```diff
@@ -0,0 +1,27 @@
+const twilio = require('twilio');
+
+const AccessToken = twilio.jwt.AccessToken;
+const ChatGrant = AccessToken.ChatGrant;
+
+exports.TokenGenerator = function (identity, deviceID) {
+    // Create a unique ID for the client on their current device
+    const endpointId = `TwilioChat:${identity}:${deviceID}`;
+
+    // Create a "grant" which enables a client to use Chat as a given user,
+    // on a given device
+    const chatGrant = new ChatGrant({
+        serviceSid: process.env.TWILIO_CHAT_SERVICE_SID,
+        endpointId,
+    });
+    // Create an access token which we will sign and return to the client,
+    // containing the grant we just created
+    const token = new AccessToken(
+        process.env.TWILIO_ACCOUNT_SID,
+        process.env.TWILIO_API_KEY,
+        process.env.TWILIO_API_SECRET,
+    );
+    token.addGrant(chatGrant);
+    token.identity = identity;
```

Comment:
When I was just using this, and not force casting the identity to a string, I was seeing the identity field show up empty, and getting an error from Twilio saying there was no identity in the token

Reply:
Interesting. Anything in Javascript can be converted to a string so you think this would just work. Was the error on the client side or the server side?

Reply:
Yeah I was surprised that it didn't work. The actual error was returned from the Twilio server, but I added api-side logs and client side logs to see the token in both places, and it was true that even after setting that identity field, the identity was still showing up empty in the token.